blogposts/2021/how-to-not-break-a-search-engine-or-what-i-learned-about-unglamorous-engineering.md

```diff
@@ -0,0 +1,297 @@
+cascade into out-of-band releases for our distribution team and upgrades for
+customers. They also tend to have a latent effect on engineer productivity (in
+this case, mine) when bugs later impose context-switches to fix
+things—conceptually big costs that I wanted to avoid. All of these
+considerations, code changes, and heaps of testing happened in the pursuit of
+an unglamorous outcome while two, maybe three, engineers reviewed the code to
+see it play out. I know this isn't an isolated thing. I get faint hints of
+other engineers at Sourcegraph doing momentous but unglamorous things that most
+of the organization is blissfully unaware of. And the Twitterverse suggests
+there's more of it happening in software all around us:
+
+<blockquote class="twitter-tweet tw-align-center"><p lang="en" dir="ltr">A huge problem in software companies is that large new features get praise, promotions, accolades... while migrating off a legacy system, increasing performance 2,4,10X, or reducing error rates, pages, or alerts by X% is often only recognized by peers and not leadership.</p>&mdash; Dan Mayer (@danmayer) <a href="https://twitter.com/danmayer/status/1395564252308541440?ref_src=twsrc%5Etfw">May 21, 2021</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+I empathize with the engineers who don't have an audience for their unglamorous
+work, who want to say "I did A Thing, there's nothing to see, but more people
+should care. Let me tell you about it!"  I like my portion of showpiece
+engineering, don't get me wrong. But shouldn't doing the necessary, unglamorous
+work be a marketable skill as well? Where's the signage that reads "Unglamorous
+engineers wanted. Will pay handsomely"? I hope you're encouraged to share what
+you've pulled off.
+
+---
+
+<a id="footnote-1"><sup>1</sup><small>If you've been following along, here's
+where our three-course meal analogy breaks down. Let's just say
+differential testing is the surprise food inspector hanging out in the
+kitchen.</small></a>
+
+<a id="footnote-2"><sup>2</sup><small>Amusingly, I once met an engineer from a
+well-known tech company who used this title on their business card.</small></a>
+
+---
+
+**About the author**
+
+[Rijnard](https://twitter.com/rvtond) is interested in developing new ways to
+search, manipulate, and fix code.  He holds a PhD in Computer Science from
+Carnegie Mellon University, where he researched automated bug fixing. He enjoys
+fundamental research but also wants to make research ideas a reality in
+practice. That's why he currently works at Sourcegraph, where he applies his
+research background to develop new tools and techniques for large-scale code
+search and automated refactoring.
+
+<small>Acks: Thanks [Rebecca Dodd](/company/team#rebecca-dodd-she-her) and [Camden Cheek](https://twitter.com/camdendcheek) for feedback on the content of this post.</small>
```

Comment:
@rebeccadee @camdencheek please check you are OK with this and if so, that links point to where you prefer :-)

Reply:
Looks great -- thanks! 😄